src/sentry/search/snuba/executors.py

```diff
@@ -1837,6 +1843,22 @@ def query(
                     where_conditions.append(
                         Condition(Column("occurrence_type_id", joined_entity), Op.IN, group_types)
                     )
+                else:
+                    all_group_type_objs: list[GroupType] = [
+                        gt_registry.get_by_type_id(id)
+                        for id in gt_registry.get_all_group_type_ids()
```

Comment:
I think we should do this like this, on line 1828
```
if not is_errors:
    visible_group_types = group_types_from(search_filters)
    if visible_group_types is None:
        # When no group types are specified, default to all default group types
        visible_group_types = [gt for gt in all_group_type_objs if not gt.in_default_search]
    # remove `raw_group_types is not None` check, since it'll never be none now, just potentially empty
    # Then just run the same logic we were running before
```

This keeps the logic shared which should reduce the chance of bugs. Since in your new branch you missed `grouptype.registry.get_visible`, and so we'd have started showing unreleased issue types to all users.

Reading this code, this might actually be a bug already... if no issue types are specified, we're not filtering to visible issue types only. So doing this will fix the bug

Reply:
cool, this line should be like this right?
[gt for gt in all_group_type_objs if ~~not~~ gt.in_default_search]

Reply:
the visible filter seems to happen in the if not None case

Reply:
Moved the logic to the base case of `group_types_from`. For some reason this causes test_snuba_perf_issue to fail, not sure why, it worked once locally so could be flakey.

Reply:
For the test, it's because the GroupType created by storing a transaction, 2001 ProfileFileIOGroupType, is not registered. 

Not sure how GroupTypes are registered in prod, but this change makes it so issues with unregistered GroupTypes are never listed (before we wouldn't filter on type).

Reply:
> this change makes it so by default, issues with unregistered GroupTypes are never listed (before, we would just skip the filter).

Do we expect all group types to be registered in prod? Is it ok to manually register the 2001 type to pass this test?

Reply:
Not totally sure what you mean by it not being registered? It might just be because the group type hasn't been marked as visible?

You can do that like https://github.com/getsentry/sentry/blob/9d2507844acf478444d3b2bc9df1c66a3ae8ca5f/tests/snuba/search/test_backend.py#L3583

Reply:
ohh got it, that was it, thanks!